### DISCUSSION TITLE
is there any way for Result.Tap for async method calls?

### DISCUSSION THREAD

**Discussion by nagtilaklaxman** [vkhorikov/CSharpFunctionalExtensions#375] [Q&A] [answered]
is there any way for Result.Tap for async method calls?

**Comment by hankovich** [comment, marked as answer]
Sure, there are tons of Tap extensions. They are located in the following files:
https://github.com/vkhorikov/CSharpFunctionalExtensions/blob/master/CSharpFunctionalExtensions/Result/Extensions/Tap.cs
https://github.com/vkhorikov/CSharpFunctionalExtensions/blob/master/CSharpFunctionalExtensions/Result/Extensions/TapAsyncBoth.cs
https://github.com/vkhorikov/CSharpFunctionalExtensions/blob/master/CSharpFunctionalExtensions/Result/Extensions/TapAsyncLeft.cs
https://github.com/vkhorikov/CSharpFunctionalExtensions/blob/master/CSharpFunctionalExtensions/Result/Extensions/TapAsyncRight.cs

In short, you can call `Tap` on `Result`/`Result<T>`/`Result<T, TError>`/`UnitResult<TError>` and on `Task<Result>`/`Task<Result<T>>`/`Task<Result<T, TError>>`/`Task<UnitResult<TError>>` passing `Action`/`Action<T>`/`Func<Task>`/`Func<T, Task>`.

So yes, async methods are definitely supported!
